### ISSUE TITLE
7.14.0: no-unused-prop-types throws exception on intermediate destructuring props 

### ISSUE THREAD

**Issue by johnhunter** [yannickcr/eslint-plugin-react#2323]
Version 7.14.0

Using intermediate destructuring of props in a class render method throws an exception. This seems to have been introduced in the [v7.14.0 release](https://github.com/yannickcr/eslint-plugin-react/releases/tag/v7.14.0)

The following fails with `VariableDeclarator ASTNodes are not handled by markPropTypesAsUsed`:
```
  render() {
    const { chartData, chartTitle, loading, className } = this.props;
    const { legendData, title, tooltip } = chartTitle;
  ...
```
Whereas the following passes:
```
  render() {
    const {
      chartData,
      chartTitle: { legendData, title, tooltip },
      loading,
      className
    } = this.props;
  ...
```
Relevant propTypes are:
```
chartData: PropTypes.array.isRequired,
chartTitle: PropTypes.object.isRequired,
loading: PropTypes.bool.isRequired,
className: PropTypes.string
```

I didn't expect the intermediate destructuring to fail, but if it is invalid then it should fail cleanly rather than throw.

**Comment by golopot** [comment]
Duplicate of #2319. Fix is merged in master but waiting for release.

**Comment by yannickcr** [comment]
Just released `v7.14.1` with the fix :slightly_smiling_face:

**Comment by johnhunter** [comment]
Great! thanks.
